utils/frame/try-runtime/cli/src/commands/offchain_worker.rs

```diff
@@ -0,0 +1,158 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+	build_executor, ensure_matching_spec_name, extract_code, full_extensions, hash_of,
+	local_spec_name, parse, state_machine_call, SharedParams, State, LOG_TARGET,
+};
+use parity_scale_codec::Encode;
+use remote_externalities::rpc_api;
+use sc_executor::NativeExecutionDispatch;
+use sc_service::Configuration;
+use sp_core::storage::well_known_keys;
+use sp_runtime::traits::{Block as BlockT, Header, NumberFor};
+use std::{fmt::Debug, str::FromStr};
+
+/// Configurations of the [`Command::OffchainWorker`].
+#[derive(Debug, Clone, structopt::StructOpt)]
+pub struct OffchainWorkerCmd {
+	/// Overwrite the wasm code in state or not.
+	#[structopt(long)]
+	overwrite_wasm_code: bool,
+
+	/// The block hash at which to fetch the block.
+	///
+	/// If the `live` state type is being used, then this can be omitted, and is equal to whatever
+	/// the `state::at` is. Only use this (with care) when combined with a snapshot.
```

Comment:
Is this still required for use with a snapshot?

Reply:
Also if this required with snapshot can we use a required_if

Reply:
excellent point. Same applies to `block_at` as well.

Reply:
ahh wait, but after a lot of hours of investigation and discussion on gh, I realized that you can't do `required_if` with a subcommand. 

Note that I manually implement the same thing now in the implementation of `fn header_at`, so it is basically the same thing.